internal/pkg/template/input.go

```diff
@@ -0,0 +1,77 @@
+package template
+
+import (
+	"reflect"
+
+	goValidator "github.com/go-playground/validator/v10"
+	"github.com/umisama/go-regexpcache"
+
+	"github.com/keboola/keboola-as-code/internal/pkg/utils/orderedmap"
+	"github.com/keboola/keboola-as-code/internal/pkg/validator"
+)
+
+type Inputs []*Input
+
+type Input struct {
+	Id          string                 `json:"id" validate:"required,template-input-id"`
+	Name        string                 `json:"name" validate:"required"`
+	Description string                 `json:"description" validate:"required"`
+	Type        string                 `json:"type" validate:"required"`
```

Comment:
Obmedzil by som typy co podporujeme, aby bolo jasne co sa moze zadat.
`bool,string,int,float64,[]string,[]int,[]float64` ?

Alebo este lepsie, vymysliet ako to suvisi s `Kind`.

Myslim, ze je to nejak takto:
```
input - string/int/float64
password  - string
textarea - string
confirm - bool
select - string
multiselect - []string
```

Sorry, ze to komplikujem, ale ta validacia je klucova, ... aby sme sa na to mohli v dalsom kode spolahnut.
Validovat podla `Kind` povolene `Type` by nemuselo byt zlozite? (kludne dalsi PR).

Mozno teda spravit `Type` volitelny? Alebo vyzadovany iba pre `input`?

--------------------- 
TLDR: teda potrebujeme zabezpecit, aby sa nedala zadat nevalidna kombinacia `Type/Kind`, ... riesenie je na tebe.

Reply:
Jo to je pravda, password a textarea bude vždycky string. Akorát (multi)select by mohl být imho i int?

Reply:
Teoreticky ano ... len na nenapadla ziadna hodnota, ... kde by si potreboval vybrat viacero cisel (alebo jedno cislo).
Mozes dat aj `[]int`. ...  a potom aj `[]float` .... len nasledne musis zlozitejsie validovat `options` a `default` - pri `select` a `multiselect`.

Reply:
ok, dám zatím string a případně se to může rozšířit. 🙂